palace/linalg/rap.cpp

```diff
@@ -49,21 +50,23 @@ void ParOperator::SetEssentialTrueDofs(const mfem::Array<int> &tdof_list,
 void ParOperator::EliminateRHS(const Vector &x, Vector &b) const
 {
   MFEM_VERIFY(A, "No local matrix available for ParOperator::EliminateRHS!");
-  auto &lx = trial_fespace.GetLVector<Vector>();
-  auto &ly = GetTestLVector();
+  auto lx = workspace::NewVector<Vector>(trial_fespace.GetVSize());
   {
-    auto &tx = trial_fespace.GetTVector<Vector>();
+    auto tx = workspace::NewVector<Vector>(trial_fespace.GetTrueVSize());
     tx = 0.0;
-    linalg::SetSubVector(tx, dbc_tdof_list, x);
+    linalg::SetSubVector<Vector>(tx, dbc_tdof_list, x);
     trial_fespace.GetProlongationMatrix()->Mult(tx, lx);
   }
 
   // Apply the unconstrained operator.
+  auto ly = workspace::NewVector<Vector>(test_fespace.GetVSize());
   A->Mult(lx, ly);
 
-  auto &ty = test_fespace.GetTVector<Vector>();
-  RestrictionMatrixMult(ly, ty);
-  b.Add(-1.0, ty);
+  {
```

Comment:
`ly` can be moved into the limited scope too.

Reply:
This scope block is actually not necessary I think since `ly` and `ty` must always be present in the same scope. So I've removed in f3ef607b to be more similar to the rest of the `ParOperator` functions.